### PR TITLE
Feat: add timestamp to race json

### PR DESCRIPTION
**Pull request by anthonyjpesce**
## Description

This PR adds the Clarity `Result` object timestamp to the exported json for each `Race`. Expected output is something like:

```
{
  "President of the United States": {
    "source": "clarity",
    "timestamp": "2020-11-06 13:05:50-05:00",
    "name": "President of the United States",
    "precinctsReportingPct": 100.0,
...
```
I've also added `python-dateutil` as a dependency for the string -> datetime -> string conversion.

## Jira Ticket

None, related to: https://arcpublishing.atlassian.net/browse/ELECT-1599

## Test Steps

`$ pip install -e .` and run something like `elexclarity 105369 GA --level=precinct --filename="tests/fixtures/atkinson_precincts_11-3.xml"`

### CODE REVIEW

File: src/elexclarity/formatters.py [hunk 1]
```diff
@@ -125,6 +124,7 @@ def transform_contest(self, contest, level, fips):
 
         return {
             "source": "clarity",
+            "timestamp": timestamp,
```
Comment (st421) [inline src/elexclarity/formatters.py:127]: `lastUpdated` is the field we use

File: tests/test_convert.py [hunk 2]
```diff
@@ -51,6 +51,7 @@ def test_format_atkinson_precincts(atkinson_precincts):
     assert len(results.keys()) == 25
     assert "President of the United States" in results.keys()
     assert results["President of the United States"]["precinctsReportingPct"] == 100
+    assert results["President of the United States"]["timestamp"] == "2020-11-06 13:05:50-05:00"
```
Comment (st421) [inline tests/test_convert.py:54]: We need this to be in the format `2020-11-06T13:05:50Z`

File: src/elexclarity/formatters.py (hunk 1 above)
Comment (anthonyjpesce) [reply src/elexclarity/formatters.py]: fixed

File: tests/test_convert.py (hunk 2 above)
Comment (anthonyjpesce) [reply tests/test_convert.py]: should be better now, but you should probably double check the logic: https://github.com/washingtonpost/elex-clarity/pull/6/files#diff-de97c87d907e2bf3af5b2030985c1683dc38eb2b101ce532a007fc5bd6247ff5R143

File: src/elexclarity/formatters.py [hunk 3]
```diff
@@ -136,16 +136,22 @@ def transform_result_object(self, result, level):
         Transforms a Clarity `Result` object into our expected format.
         """
         fips = None
+
+        # convert the timestamp and make sure we're in EST
+        est = tz.gettz("America/New_York")
+        timestamp = parser.parse(result["Timestamp"], tzinfos={"EST": est}).astimezone(est)
+        timestamp = timestamp.strftime("%Y-%m-%dT%H:%H:%SZ")
```
Comment (st421) [inline src/elexclarity/formatters.py:143]: `"%Y-%m-%dT%H:%M:%SZ"` I think

Comment (anthonyjpesce) [reply src/elexclarity/formatters.py]: Ahem... wow. Should be fixed

**Review by st421 — APPROVED** [review]

**Pull request merged**